runtime/neumann/src/fee_handler.rs

```diff
@@ -0,0 +1,163 @@
+	xcm_data: Option<XcmCurrencyData>,
+	asset_metadata: Option<AssetMetadata<Balance, CustomMetadata>>,
+}
+impl Default for FeeInformation {
+	fn default() -> FeeInformation {
+		FeeInformation { token_id: NATIVE_TOKEN_ID, xcm_data: None, asset_metadata: None }
+	}
+}
+pub trait CallParser<Call> {
+	fn fee_information(call: &Call) -> FeeInformation;
+}
+pub struct FeeCallParser;
+impl CallParser<Call> for FeeCallParser {
+	fn fee_information(c: &Call) -> FeeInformation {
+		if let Call::AutomationTime(pallet_automation_time::Call::schedule_xcmp_task {
+			para_id,
+			currency_id,
+			..
+		}) = c.clone()
+		{
+			let xcm_data = XcmpHandler::get_xcm_chain_data(u32::from(para_id), currency_id);
+			let asset_metadata = AssetRegistry::metadata(currency_id);
+
+			FeeInformation { token_id: currency_id, xcm_data, asset_metadata }
+		} else {
+			FeeInformation::default()
+		}
+	}
+}
+pub type CallOf<T> = <T as frame_system::Config>::Call;
+type NegativeImbalanceOf<C, T> =
+	<C as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
+
+pub struct DuplicateCurrencyAdapter<MC, C, OU, TA, FCP>(PhantomData<(MC, C, OU, TA, FCP)>);
+
+impl<T, MC, C, OU, TA, FCP> OnChargeTransaction<T> for DuplicateCurrencyAdapter<MC, C, OU, TA, FCP>
+where
+	T: pallet_transaction_payment::Config,
+	C: Currency<<T as frame_system::Config>::AccountId>,
+	C::Balance: From<MC::Balance>,
+	C::PositiveImbalance: Imbalance<
+		<C as Currency<<T as frame_system::Config>::AccountId>>::Balance,
+		Opposite = C::NegativeImbalance,
+	>,
+	C::NegativeImbalance: Imbalance<
+		<C as Currency<<T as frame_system::Config>::AccountId>>::Balance,
+		Opposite = C::PositiveImbalance,
+	>,
+	MC::CurrencyId: From<TokenId>,
+	MC::Balance: From<C::Balance>,
+	MC: MultiCurrency<<T as frame_system::Config>::AccountId>,
+	OU: OnUnbalanced<NegativeImbalanceOf<C, T>>,
+	TA: Get<T::AccountId>,
+	FCP: CallParser<CallOf<T>>,
+{
+	type LiquidityInfo = Option<NegativeImbalanceOf<C, T>>;
+	type Balance = <C as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+	/// Withdraw the prxedicted fee from the transaction origin.
+	///
+	/// Note: The `fee` already includes the `tip`.
+	fn withdraw_fee(
+		who: &T::AccountId,
+		call: &T::Call,
+		_info: &DispatchInfoOf<T::Call>,
+		fee: Self::Balance,
+		tip: Self::Balance,
+	) -> Result<Self::LiquidityInfo, TransactionValidityError> {
+		if fee.is_zero() {
+			return Ok(None)
+		}
+
+		// cross it
+		let call_information = FCP::fee_information(call.clone());
+
+		// TODO: when no info or we don't allow that token throw error
+
+		if call_information.token_id == NATIVE_TOKEN_ID {
+			let withdraw_reason = if tip.is_zero() {
+				WithdrawReasons::TRANSACTION_PAYMENT
+			} else {
+				WithdrawReasons::TRANSACTION_PAYMENT | WithdrawReasons::TIP
+			};
+
+			match C::withdraw(who, fee, withdraw_reason, ExistenceRequirement::KeepAlive) {
+				Ok(imbalance) => Ok(Some(imbalance)),
+				Err(_) => Err(InvalidTransaction::Payment.into()),
+			}
```

Comment:
Code should match what occurs in  the `pallet_transaction_payment` `CurrencyAdapter`.